6_predict/src/plot_diagnostics.R

```diff
@@ -447,6 +447,101 @@ make_class_prediction_map <- function(class_probs, reaches, out_dir,
   return(fnames)
 }
 
+
+make_transition_region_map <- function(class_probs, reaches,
+                                       region_threshold,
+                                       out_dir, model_name, pt_size = 0.5) {
+  #' @description this function creates a map showing the transition regions 
+  #' (reaches not distinctly predicted in single cluster)
+  #' 
+  #' @param class_probs dataframe of predicted class probabilities for each reach.
+  #' must have an "ID" column and columns for the class probabilities labeled with
+  #' the name of the class. No other columns. 
+  #' @param reaches sf object containing the reaches to plot. Must have a "COMID" column
+  #' @param out_dir where output figures are saved
+  #' @param region_threshold minimum probability for region to be considered transitional
+  #' @param model_name name to add to the file name that describes this model
+  #' @param ncores number of cores to use
+  #' @param pt_size size of features to be mapped
+  #' 
+  #' @return file paths to maps
+  
+  #identify region rankings based on probability
+  #identify region rankings based on probability
+  region_probs <- select(class_probs, -ID) %>%
+    rename(prob_1 = 1, prob_2 = 2, prob_3 = 3, prob_4 = 4, prob_5 = 5)
+  region_ranks <- t(apply(-region_probs, 1, rank, ties.method = "random")) %>% 
```

Comment:
I made a `decreasing_rank` [function](https://github.com/USGS-R/regional-hydrologic-forcings-ml/blob/2c84ee34203473f3a0893e7b9ef8a320f87457ec/6_predict/src/plot_diagnostics.R#L471) that could be used here instead of rank of the negative values 

Also, if this takes a long time, could run with parApply instead. I used that for the other CONUS NHD maps